nsv13/code/modules/overmap/weapons/mines.dm

```diff
@@ -0,0 +1,99 @@
+/obj/structure/space_mine
+	name = "space mine"
+	desc = "Like a naval mine, but in space!"
+	icon = 'nsv13/icons/overmap/effects.dmi'
+	icon_state = "mine_syndicate"
+	anchored = TRUE
+	density = FALSE
+	layer = ABOVE_MOB_LAYER
+	animate_movement = NO_STEPS
+	max_integrity = 300
+	integrity_failure = 100
+	var/datum/star_system/current_system
+	var/faction = "syndicate" //evil mines
+	var/damage = 100
+	var/damage_type = BRUTE
+	var/damage_flag = "overmap_heavy"
+	alpha = 50 //They're supposed to be sneaky, their main advantage is being cloaked
+
+/obj/structure/space_mine/Initialize(mapload, var/list/coordinates, var/new_faction, var/datum/star_system/system)
+	. = ..()
+	if(system)
+		current_system = system
+	else //Someone is probably spawning us on the overmap, so we assume it's next to the main ship
+		current_system = SSstar_system.find_main_overmap().current_system
+	current_system.system_contents |= src
+	if(new_faction)
+		faction = new_faction
+	update_icon()
+	if(coordinates)
+		x = coordinates["x"]
+		y = coordinates["y"]
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/space_mine/Destroy(force)
+	current_system?.contents_positions.Remove(src)
+	current_system?.system_contents.Remove(src)
+	RemoveElement(/datum/element/connect_loc)
+	. = ..()
+
+/// This makes us not drift like normal objects in space do
+/obj/structure/space_mine/Process_Spacemove(movement_dir = 0)
+	return 1
+
+/obj/structure/space_mine/proc/on_entered(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+
+	if(!(istype(AM, /obj/structure/overmap) || istype(AM, /obj/item/projectile)))
+		return
+
+	switch(AM.type)
+		if(/obj/item/projectile)
+			var/obj/item/projectile/P = AM
+			if(P.faction != faction || !(P.faction == "nanotrasen" || P.faction == "solgov") && (faction == "nanotrasen" || faction == "solgov"))
+				P.Impact(src)
+
+		if(/obj/structure/overmap)
+			var/obj/structure/overmap/OM = AM
+			if(OM.faction != faction || !(OM.faction == "nanotrasen" || OM.faction == "solgov") && (faction == "nanotrasen" || faction == "solgov"))
+				mine_explode(OM)
+
+/obj/structure/space_mine/update_icon(updates)
+	. = ..()
+	icon_state = "mine_[faction]"
```

Comment:
Should have a default icon if there's no proper faction icon. Maybe checked with a static list of which icons you do have.
Potentially unsafe otherwise.